code_style.md

```diff
@@ -225,6 +225,11 @@ Unless otherwise specified, the following applies to all code:
     }
     ```
 
+37. Avoid functions whose behaviour / return type varies with different parameter types.
+    Multiple return types are fine when appropriate (eg. SDKConfig.get() with a string param which
+    returns the type according to the param given) but SDKConfig.get() with no args returning the
+    whole config object is not: this could just be a separate function.
```

Comment:
It would be nice to define "when appropriate" a bit better.

```suggestion

    Multiple return types are acceptable for operations such as a lookup based on a string parameter
    `SDKConfig.get()`, whose return type varies according to the `key` parameter, is an example of this.
    
    However, completely changing the behaviour of the function based on the parameter types is not
    acceptable: instead, use a separate function. For example, it would not be acceptable for
    `SDKConfig.get()` with no arguments return the whole configuration object.
```

Reply:
Better hopefully?